addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/handler/ModbusDataThingHandler.java

```diff
@@ -752,9 +758,13 @@ public synchronized void onWriteResponse(ModbusWriteRequestBlueprint request, Mo
             return;
         }
         logger.debug("Successful write, matching request {}", request);
-        DateTimeType now = new DateTimeType();
-        updateStatus(ThingStatus.ONLINE);
-        updateState(ModbusBindingConstantsInternal.CHANNEL_LAST_WRITE_SUCCESS, now);
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
```

Comment:
Or should the framework be optimized better, @ssalonen how expensive was this?

Reply:
The general suggestion is to throttle it in the binding this way if there are status updates every few seconds. I guess that's the case for modbus, so this optimisation is fine. We could also think about moving something like that to the framework, but having it transparent in the binding is imho not bad.

Reply:
Great thanks 👍

Reply:
The LIFX binding had a similar issue which was addressed with https://github.com/eclipse/smarthome/pull/3816. It sends the same update at most once a second. That way it can still be used for the "last seen online" feature that will be there some day. :wink:

Reply:
Nice find @wborn. I will adapt the code to follow the convention set by LIFX.